src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -997,6 +997,23 @@ TargetListOnPartitionColumn(Query *query, List *targetEntryList)
 		Expr *targetExpression = targetEntry->expr;
 
 		bool isPartitionColumn = IsPartitionColumn(targetExpression, query);
+		Oid relationId = InvalidOid;
+		Var *column = NULL;
+
+		FindReferencedTableColumn(targetExpression, NIL, query, &relationId, &column);
+
+		/*
+		 * If the expression belongs to reference table directly returns true,
```

Comment:
also function header needs revisiting since the function performs differently now